common/src/main/java/io/netty/util/AsciiString.java

```diff
@@ -1331,155 +1016,70 @@ public boolean contains(CharSequence cs) {
     }
 
     public long parseLong(int start, int end, int radix) {
-        if (radix < Character.MIN_RADIX || radix > Character.MAX_RADIX) {
-            throw new NumberFormatException();
-        }
-
-        if (start == end) {
-            throw new NumberFormatException();
-        }
-
-        int i = start;
-        boolean negative = charAt(i) == '-';
-        if (negative && ++i == end) {
-            throw new NumberFormatException(subSequence(start, end).toString());
-        }
-
-        return parseLong(i, end, radix, negative);
+        return parseAsciiLong(start, end, radix);
     }
 
-    private long parseLong(int start, int end, int radix, boolean negative) {
-        final byte[] value = this.value;
-        long max = Long.MIN_VALUE / radix;
-        long result = 0;
-        int offset = start;
-        while (offset < end) {
-            int digit = Character.digit((char) (value[offset++] & 0xFF), radix);
-            if (digit == -1) {
-                throw new NumberFormatException(subSequence(start, end).toString());
-            }
-            if (max > result) {
-                throw new NumberFormatException(subSequence(start, end).toString());
-            }
-            long next = result * radix - digit;
-            if (next > result) {
-                throw new NumberFormatException(subSequence(start, end).toString());
-            }
-            result = next;
-        }
-        if (!negative) {
-            result = -result;
-            if (result < 0) {
-                throw new NumberFormatException(subSequence(start, end).toString());
-            }
-        }
-        return result;
+    public char parseChar(int start) {
+        return charAt(start);
     }
 
     public short parseShort() {
-        return parseShort(0, length(), 10);
+        return parseAsciiShort();
     }
 
     public short parseShort(int radix) {
-        return parseShort(0, length(), radix);
+        return parseAsciiShort(radix);
     }
 
     public short parseShort(int start, int end) {
-        return parseShort(start, end, 10);
+        return parseAsciiShort(start, end);
     }
 
     public short parseShort(int start, int end, int radix) {
-        int intValue = parseInt(start, end, radix);
-        short result = (short) intValue;
-        if (result != intValue) {
-            throw new NumberFormatException(subSequence(start, end).toString());
-        }
-        return result;
+        return parseAsciiShort(start, end, radix);
     }
 
     public float parseFloat() {
-        return parseFloat(0, length());
+        return parseAsciiFloat();
     }
 
     public float parseFloat(int start, int end) {
-        return Float.parseFloat(toString(start, end));
+        return parseAsciiFloat(start, end);
     }
 
     public double parseDouble() {
-        return parseDouble(0, length());
+        return parseAsciiDouble();
     }
 
     public double parseDouble(int start, int end) {
```

Comment:
This will make IDE show `parseDouble()` and `parseAsciiDouble()` which do the same job. Could we just remove the ones at `AsciiString`? Please also port this change to 4.1 if we have the same issue there.